libtiny_client/src/state.rs

```diff
@@ -232,6 +271,31 @@ impl StateInner {
                 }
             }
 
+            // Reply 477 when user needs to be identified with NickServ to join a channel
+            Reply { num: 477, params } => {
+                // Only try to automatically rejoin if nickserv_ident is configured
+                if self.nickserv_ident.is_some() {
+                    // Get channel name
+                    let channel = params.get(1).unwrap().clone();
```

Comment:
This unwrap can potentially cause a crash on non-standard-confirming servers (probably on irc.gitter.im for example). Instead we should pattern match on `params.get(1)`, log it if it's `None`.